fail if python version is < 3.5

f0 opened the pull request:
python >= 3.5 is needed for  json.decoder.JSONDecodeError

fkoch@linux-bhu4:~/projects/tmp> orca-build -t a  .
Traceback (most recent call last):
  File "/usr/bin/orca-build", line 296, in parse
    args = json.loads(rest.strip())
  File "/usr/lib64/python3.4/json/__init__.py", line 318, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.4/json/decoder.py", line 343, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.4/json/decoder.py", line 361, in raw_decode
    raise ValueError(errmsg("Expecting value", s, err.value)) from None
ValueError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/orca-build", line 742, in <module>
    __wrapped_main__()
  File "/usr/bin/orca-build", line 740, in __wrapped_main__
    main(ctx, config)
  File "/usr/bin/orca-build", line 705, in main
    builder = Builder(ctx, build_args=config.build_args, rootless=config.rootless)
  File "/usr/bin/orca-build", line 356, in __init__
    self.script = DockerfileParser(contents).parse()
  File "/usr/bin/orca-build", line 297, in parse
    except json.decoder.JSONDecodeError:
AttributeError: 'module' object has no attribute 'JSONDecodeError'

cyphar commented:
You need to add a `Signed-off-by:` line to your commit(s) which indicates that you attest the [Developer Certificate of Origin](https://developercertificate.org/) a statement about your contributions that you *must read* before signing (don't worry, it's quite short and easy-to-read). You can add it to your commits with `git commit --amend -s`, and then doing a `git push --force`.

f0 committed:
fail if python version is < 3.5

python >= 3.5 is needed for  json.decoder.JSONDecodeError

fkoch@linux-bhu4:~/projects/tmp> orca-build -t a  .
Traceback (most recent call last):
  File "/usr/bin/orca-build", line 296, in parse
    args = json.loads(rest.strip())
  File "/usr/lib64/python3.4/json/__init__.py", line 318, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.4/json/decoder.py", line 343, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.4/json/decoder.py", line 361, in raw_decode
    raise ValueError(errmsg("Expecting value", s, err.value)) from None
ValueError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/orca-build", line 742, in <module>
    __wrapped_main__()
  File "/usr/bin/orca-build", line 740, in __wrapped_main__
    main(ctx, config)
  File "/usr/bin/orca-build", line 705, in main
    builder = Builder(ctx, build_args=config.build_args, rootless=config.rootless)
  File "/usr/bin/orca-build", line 356, in __init__
    self.script = DockerfileParser(contents).parse()
  File "/usr/bin/orca-build", line 297, in parse
    except json.decoder.JSONDecodeError:
AttributeError: 'module' object has no attribute 'JSONDecodeError'

Signed-off-by: Florian Koch <florian.koch1981@gmail.com>
[ pep8 fixes ]
Signed-off-by: Aleksa Sarai <asarai@suse.de>

f0 commented:
@cyphar   done

PI-Victor commented:
just ran into this and while it's trivial, it would be nice to have a warning, @cyphar can you merge it?

cyphar commented:
LGTM.
